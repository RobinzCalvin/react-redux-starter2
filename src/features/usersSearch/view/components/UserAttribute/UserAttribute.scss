.user-attribute {
  display: flex;
  align-items: center;
  justify-content: space-between;
  padding: 0 1rem;
  height: 2.5rem;
  border-bottom: 0.15rem solid #e5e5e5;
  color: #000;
  text-decoration: none;

  &:hover {
    border-color: #bbb;
  }
}
