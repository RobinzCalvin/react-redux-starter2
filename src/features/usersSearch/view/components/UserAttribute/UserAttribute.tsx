import React from 'react';
import block from 'bem-cn';

import './UserAttribute.scss';

interface IProps {
  URL: string;
  title: string;
  value: number;
}

const b = block('user-attribute');

function UserAttribute(props: IProps) {
  const { title, value, URL } = props;
  return (
    <a href={URL} target="_blank" className={b()}>
      {title}
      <span className={b('value')}>{value}</span>
    </a>
  );
}

export default UserAttribute;
