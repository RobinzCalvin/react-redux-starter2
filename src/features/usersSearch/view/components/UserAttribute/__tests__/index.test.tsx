import React from 'react';
import { shallow } from 'enzyme';

import UserAttribute, { IUserAttributeProps } from '../UserAttribute';

const props: IUserAttributeProps = {
  title: 'title',
  URL: 'https://the.url',
  value: 11,
};

describe('UserAttribute component', () => {
  const component = shallow(<UserAttribute {...props} />);
  it('should match snapshot', () => {
    expect(component).toMatchSnapshot();
  });
});
