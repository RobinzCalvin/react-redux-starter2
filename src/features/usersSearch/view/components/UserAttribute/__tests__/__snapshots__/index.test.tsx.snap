// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UserAttribute component should match snapshot 1`] = `
<a
  className="user-attribute"
  href="https://the.url"
  target="_blank"
>
  title
  <span
    className={
      Object {
        "has": [Function],
        "is": [Function],
        "mix": [Function],
        "split": [Function],
        "state": [Function],
        "toString": [Function],
      }
    }
  >
    11
  </span>
</a>
`;
